groundstation/backend_api/user.py

```diff
@@ -83,15 +95,63 @@ def patch(self, auth_token, local_data=None):
             setattr(user, attribute, validated_data[attribute])
 
         user.regenerate_password_hash(post_data.get('password'))
-
-        db.session.commit()
+        try:
+            db.session.commit()
+        except exc.IntegrityError as e:
+            db.session.rollback()
+            # TODO: Probably remove dev_message
+            response_object = {
+                'status':'fail',
+                'message':'Username already taken!',
+                'dev_message':str(e.orig)
+            }
+            return response_object, 400
 
         response_object = {
             'status': 'success',
             'data': user.to_json()
         }
         return response_object, 200
 
+    @create_context
+    @login_required
+    def delete(self, auth_token, local_data=None):
```

Comment:
Need to add some checks to this rather powerful endpoint:

- [ ] Only admins can delete a user
- [ ] The to-be-deleted user must be created by the admin calling this endpoint
- [ ] Admin users cannot be deleted
    - I don't imagine us needing to delete admin users and it might get messy if were trying to delete an admin that has created users (and what if one of those users is also an admin that has created other users...).